p2p/src/types.rs

```diff
@@ -183,8 +183,8 @@ impl<'de> Visitor<'de> for PeerAddrs {
 				Err(_) => {
 					let socket_addrs = entry
 						.to_socket_addrs()
-						.expect(format!("Unable to resolve DNS: {}", entry).as_str());
-					peers.append(&mut socket_addrs.map(|addr| PeerAddr(addr)).collect());
+						.unwrap_or_else(|_| panic!("Unable to resolve DNS: {}", entry));
```

Comment:
can we get rid of expect/panic somehow?

Reply:
I agree that this error must be handled. Will do in a subsequent PR.